src/lib/router.ts

```diff
@@ -129,4 +85,40 @@ router.get('/matches', async (request, response, next) => {
   }
 });
 
+router.post('/highlights', async (request, response, next) => {
+  try {
+    const { timestamp, events, videoSrc, matchId } = request.body;
+
+    const requestIsInvalid =
+      typeof timestamp !== 'number' ||
+      !Array.isArray(events) ||
+      !events.every((event) => typeof event === 'string') ||
+      typeof videoSrc !== 'string' ||
+      typeof matchId !== 'string';
+
+    if (requestIsInvalid) {
+      response.status(400).send('Invalid payload');
+      return;
+    }
+
+    const highlight: Omit<MatchHighlight, '_id'> = {
+      matchId: new ObjectId(matchId),
```

Comment:
there is no validation if the matchid is a valid ObjectId, which could resolve in an exception when inserted into the query
```suggestion
    const { timestamp, events, videoSrc, matchId } = request.body;

    const requestIsInvalid =
      typeof timestamp !== 'number' ||
      !Array.isArray(events) ||
      !events.every((event) => typeof event === 'string') ||
      typeof videoSrc !== 'string' ||
      typeof matchId !== 'string';
      
     if(!ObjectId.isvalid(matchId)) {
       response.status(400).send('Invalid matchId');
       return;
     }

    if (requestIsInvalid) {
      response.status(400).send('Invalid payload');
      return;
    }

    const highlight: Omit<MatchHighlight, '_id'> = {
      matchId: new ObjectId(matchId),
```